### PR TITLE
First release of heading library.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: examples/calibration.toit [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+The calibration settings are stored in the flash with the key
+  "mag-calibration". This way, any other app on the device can take
+  advantage of the calibration settings.
+*/
+
+main:
+  store := FlashStore
+  bus := i2c.Bus
+    --sda=gpio.Pin 21
+    --scl=gpio.Pin 22
```
Comment (erikcorry) [inline examples/calibration.toit]:
```suggestion
      --sda=gpio.Pin 21
      --scl=gpio.Pin 22
```

File: examples/main.toit [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import math
+import device show FlashStore
+
+BASE_VECTOR ::= math.Point3f 1.0 0.0 0.0
+
+main:
+  store := FlashStore
+  bus := i2c.Bus
+    --sda=gpio.Pin 21
+    --scl=gpio.Pin 22
```
Comment (erikcorry) [inline examples/main.toit]:
```suggestion
      --sda=gpio.Pin 21
      --scl=gpio.Pin 22
```

File: src/heading.toit [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      vector_dot_ e_normalized base_vector
+      vector_dot_ n_normalized base_vector
+  heading := heading_rads * 180 / math.PI
+  if heading < 0.0: heading += 360.0
+  return heading
+
+vector_cross_ v1/math.Point3f v2/math.Point3f -> math.Point3f:
+  return math.Point3f
+      v1.y * v2.z - v1.z * v2.y
+      v1.x * v2.z - v1.z * v2.x
```
Comment (erikcorry) [inline src/heading.toit]: Maybe the wrong sign here?

**Review by erikcorry — APPROVED** [review]

File: examples/calibration.toit (hunk 1 above)
Comment (floitsch) [reply examples/calibration.toit]: done.

File: examples/main.toit (hunk 2 above)
Comment (floitsch) [reply examples/main.toit]: done.

File: src/heading.toit (hunk 3 above)
Comment (floitsch) [reply src/heading.toit]: Yes.
Don't think it influenced the result, though.
done.

**Pull request merged**